typing/env.ml

```diff
@@ -1103,7 +1103,7 @@ let find_module ~alias path env =
       if alias then md (fc.fcomp_res)
       else md (modtype_of_functor_appl fc p1 p2)
 
-let find_module_lazy ~alias path env =
+let find_module_lazy ?(alias=false) path env =
```

Comment:
I think that we do not need to expose this parameter in the interface and it can be left non-optional.

The key observation is that `find_strengthened_module` is the only use of `~alias:true` outside of `Env` after your refactoring, but it actually should be `~alias:false`. Since `alias` only affects the `Papply` case and `find_strengthened_module` is never called with a `Papply` path the difference is not currently detectable, but if it were called with a `Papply` path it would currently give the incorrect answer.